selfdrive/modeld/navmodeld.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+import os
+import gc
+import math
+import time
+import ctypes
+import numpy as np
+from pathlib import Path
+from typing import Tuple, Dict
+from cereal import messaging
+from cereal.messaging import PubMaster, SubMaster
+from cereal.visionipc import VisionIpcClient, VisionStreamType
+from openpilot.system.hardware import TICI
+from openpilot.system.swaglog import cloudlog
+from openpilot.common.params import Params
+from openpilot.common.realtime import set_realtime_priority
+from openpilot.selfdrive.modeld.constants import IDX_N
+from openpilot.selfdrive.modeld.models.commonmodel_pyx import Runtime
+
+USE_SNPE = int(os.getenv('USE_SNPE', str(int(TICI))))
+if USE_SNPE:
+  os.environ['ADSP_LIBRARY_PATH'] = "/data/pythonpath/third_party/snpe/dsp/"
+  from selfdrive.modeld.runners.snpemodel_pyx import SNPEModel as ModelRunner
+else:
+  from selfdrive.modeld.runners.onnxmodel_pyx import ONNXModel as ModelRunner
+
+NAV_INPUT_SIZE = 256*256
+NAV_FEATURE_LEN = 256
+NAV_DESIRE_LEN = 32
+NAV_OUTPUT_SIZE = 2*2*IDX_N + NAV_DESIRE_LEN + NAV_FEATURE_LEN
```

Comment:
Again, these constants (along with the ones in modeld and dmonitoringmodeld) should ideally be inferred by checking the input/output sizes of the onnx files, but that seemed outside the scope of this PR.